src/context/app_context.rs

```diff
@@ -1,12 +1,14 @@
 use std::collections::VecDeque;
 use std::sync::mpsc;
 
+use crate::bookmarks::BookmarkMapping;
```

Comment:
I think I would probably just store the bookmarks inside `config::AppConfig`.
Any reason you decided to have it separate?